Subject: Use kmalloc rather than kzalloc for switch log buffer
From: Jeremy Kerr <jk@ozlabs.org>
References: 447133 - LTC50070

No need to zero the entire buffer, just the head and tail indices.

Signed-off-by: Jeremy Kerr <jk@ozlabs.org>
Signed-off-by: Olaf Hering <olh@suse.de>
---
 arch/powerpc/platforms/cell/spufs/file.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/arch/powerpc/platforms/cell/spufs/file.c
+++ b/arch/powerpc/platforms/cell/spufs/file.c
@@ -2445,7 +2445,7 @@ static int spufs_switch_log_open(struct 
 		goto out;
 	}
 
-	ctx->switch_log = kzalloc(sizeof(struct switch_log) +
+	ctx->switch_log = kmalloc(sizeof(struct switch_log) +
 		SWITCH_LOG_BUFSIZE * sizeof(struct switch_log_entry),
 		GFP_KERNEL);
 
@@ -2454,6 +2454,7 @@ static int spufs_switch_log_open(struct 
 		goto out;
 	}
 
+	ctx->switch_log->head = ctx->switch_log->tail = 0;
 	init_waitqueue_head(&ctx->switch_log->wait);
 	rc = 0;
 
